pkg/controllers/runtime_controller.go

```diff
@@ -87,17 +87,17 @@ func (r *RuntimeReconciler) ReconcileInternal(ctx cruntime.ReconcileRequestConte
 		return utils.RequeueIfError(errors.Wrap(err, "Failed to create"))
 	}
 
-	// 2.Get or create the engine
-	engine, err := r.implement.GetOrCreateEngine(ctx)
+	// 2.Get the ObjectMeta of runtime
```

Comment:
Could you add comments about the reason of changing the order of step 2 and 3?

Reply:
Changing the order was used to judge GetOrCreateEngine failed reason which is runtime having deletionTimeStamp.  In this case we should ignore the GetOrCreateEngine error and continue to reconcileruntimeDeletion, but it would cause engine is a nil pointer. And we have resolvd this problem inside GetOrCreateEngine so that it would return an engine although it could not get mounted dataset. So the order is no need to change now and i will fix it.